silo-vaults/contracts/MetaMorpho.sol

```diff
@@ -322,40 +297,40 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     }
 
     /// @inheritdoc IMetaMorphoBase
-    function updateWithdrawQueue(uint256[] calldata indexes) external onlyAllocatorRole {
+    function updateWithdrawQueue(uint256[] calldata indexes) external virtual onlyAllocatorRole {
         uint256 newLength = indexes.length;
         uint256 currLength = withdrawQueue.length;
 
         bool[] memory seen = new bool[](currLength);
-        Id[] memory newWithdrawQueue = new Id[](newLength);
+        IERC4626[] memory newWithdrawQueue = new IERC4626[](newLength);
 
         for (uint256 i; i < newLength; ++i) {
             uint256 prevIndex = indexes[i];
 
             // If prevIndex >= currLength, it will revert with native "Index out of bounds".
-            Id id = withdrawQueue[prevIndex];
-            if (seen[prevIndex]) revert ErrorsLib.DuplicateMarket(id);
+            IERC4626 market = withdrawQueue[prevIndex];
+            if (seen[prevIndex]) revert ErrorsLib.DuplicateMarket(market);
             seen[prevIndex] = true;
 
-            newWithdrawQueue[i] = id;
+            newWithdrawQueue[i] = market;
         }
 
         for (uint256 i; i < currLength; ++i) {
             if (!seen[i]) {
-                Id id = withdrawQueue[i];
+                IERC4626 market = withdrawQueue[i];
 
-                if (config[id].cap != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroCap(id);
-                if (pendingCap[id].validAt != 0) revert ErrorsLib.PendingCap(id);
+                if (config[market].cap != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroCap(market);
+                if (pendingCap[market].validAt != 0) revert ErrorsLib.PendingCap(market);
 
-                if (MORPHO.supplyShares(id, address(this)) != 0) {
-                    if (config[id].removableAt == 0) revert ErrorsLib.InvalidMarketRemovalNonZeroSupply(id);
+                if (market.balanceOf(address(this)) != 0) {
+                    if (config[market].removableAt == 0) revert ErrorsLib.InvalidMarketRemovalNonZeroSupply(market);
 
-                    if (block.timestamp < config[id].removableAt) {
-                        revert ErrorsLib.InvalidMarketRemovalTimelockNotElapsed(id);
+                    if (block.timestamp < config[market].removableAt) {
+                        revert ErrorsLib.InvalidMarketRemovalTimelockNotElapsed(market);
                     }
                 }
 
-                delete config[id];
+                delete config[market];
```

Comment:
reset approval?
```suggestion
                IERC20(asset()).approve(address(market), 0);
                delete config[market];
```

Reply:
you can only delete cap when it is set to 0. I reset approval when it is wet to 0.